django/db/models/fields/__init__.py

```diff
@@ -242,31 +242,55 @@ def _check_field_name(self):
             return []
 
     def _check_choices(self):
-        if self.choices:
-            if isinstance(self.choices, str) or not is_iterable(self.choices):
-                return [
-                    checks.Error(
-                        "'choices' must be an iterable (e.g., a list or tuple).",
-                        obj=self,
-                        id='fields.E004',
-                    )
-                ]
-            elif any(isinstance(choice, str) or
-                     not is_iterable(choice) or len(choice) != 2
-                     for choice in self.choices):
-                return [
-                    checks.Error(
-                        "'choices' must be an iterable containing "
-                        "(actual value, human readable name) tuples.",
-                        obj=self,
-                        id='fields.E005',
-                    )
-                ]
-            else:
-                return []
+        if not self.choices:
+            return []
+
+        def _is_terminal(value):
+            return isinstance(value, str) or not is_iterable(value)
+
+        if _is_terminal(self.choices):
+            return [
+                checks.Error(
+                    "'choices' must be an iterable (e.g., a list or tuple).",
+                    obj=self,
+                    id='fields.E004',
+                )
+            ]
+
+        # Expect [group_name, [value, display]]
+        for choices_group in self.choices:
+            try:
+                group_name, group_choices = choices_group
+                try:
```

Comment:
Can you put 
```
except ValueError:
    # Containing non-pairs
    break
```
up here rather than nesting the next try block inside it? Tests are passing there but I'm not sure if `value, human_name = group_name, group_choices` might raise `ValueError`. Maybe a test case is missing there. If that exception is possible there, I'd opt for a separate try/except to avoid such a large one.

Reply:
I don't think `value, human_name = group_name, group_choices` can raise `ValueError`. We are just re-assigning two variables. Moving the `except ValueError` up makes the code easier to follow 👍.